src/app/components/core/configuration-manager/configuration-group/configuration-group.component.ts

```diff
@@ -0,0 +1,181 @@
+})
+export class ConfigurationGroupComponent implements OnInit {
+
+  selectedGroup = 'Default Configuration';
+  @Input() category;
+  groups = [];
+
+  @Output() changedConfigEvent = new EventEmitter<any>();
+  @Output() formStatusEvent = new EventEmitter<boolean>();
+  @Output() changedAdvanceConfigEvent = new EventEmitter<any>();
+
+  // To hold the changed configuration values of a plugin
+  configFormValues = {};
+
+  pages = ['south', 'north', 'notification'];
+  @Input() from;
+  categoryKey = '';
+
+  advanceConfiguration: any
+  securityConfiguration: any;
+  changedAdvanceConfiguration: any;
+  changedSecurityConfiguration: any;
+
+  constructor(
+    public developerFeaturesService: DeveloperFeaturesService,
+    public rolesService: RolesService,
+    private configService: ConfigurationService,
+    private configurationControlService: ConfigurationControlService,
+  ) { }
+
+  ngOnInit() { }
+
+  ngOnChanges() {
+    this.categeryConfiguration();
+    this.getChildConfigData();
+  }
+
+  public updateCategroyConfig(config) {
+    this.category.config = config;
+    this.getChildConfigData();
+    this.categeryConfiguration();
+  }
+
+  categeryConfiguration() {
+    this.groups = [];
+    const configItems = Object.keys(this.category.config).map(k => {
+      this.category.config[k].key = k;
+      return this.category.config[k];
+    });
+
+    this.groups = chain(configItems).groupBy(x => x.group).map((v, k) => {
+      if (k != "undefined") {
+        return { category: this.category.name, group: k, config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      } else {
+        return { category: this.category.name, group: "Default Configuration", config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      }
+    }).value();
+  }
+
+  /**
+   * Set configuration of the selected child category
+   * @param category Object{key, description, displayName}
+   */
+  selectTab(tab: string) {
+    if (tab !== this.selectedGroup) {
+      this.selectedGroup = tab;
+    }
+  }
+
+  public getChildConfigData() {
+    // No advance configuration on add form
+    if (this.pages.includes(this.from) && this.category) {
+      this.categoryKey = this.category.name;
+      this.checkIfAdvanceConfig(this.category.name)
+    }
+  }
+
+  checkIfAdvanceConfig(categoryName: string) {
+    this.configService.getCategoryConfigChildren(categoryName).
+      subscribe(
+        (data: any) => {
+          const categoryChildren = data.categories?.filter(cat => (cat.key == `${this.categoryKey}Advanced`) || (cat.key == `${this.categoryKey}Security`));
+          categoryChildren.forEach(cat => {
+            // Set group of advance/security configuration
+            cat.group = cat?.key.includes(`${this.categoryKey}Advanced`) ? 'Advanced Configuration' :
+              (cat?.key.includes(`${this.categoryKey}Security`) ? 'Security Configuration' : cat?.displayName);
```

Comment:
cat?.displayName ? Is this not a group name?

What if the display name is missing?

Reply:
display name is not a group name. group name is the value of the group key. For Advanced and Security configuration those are not a part of any group so we are adding group name based on the key.   

```
 const categoryChildren = data.categories?.filter(cat => (cat.key == `${this.categoryKey}Advanced`) || (cat.key == `${this.categoryKey}Security`));
          categoryChildren.forEach(cat => {
            // Set group of advance/security configuration
            cat.group = cat?.key.includes(`${this.categoryKey}Advanced`) ? 'Advanced Configuration' :
            (cat?.key.includes(`${this.categoryKey}Security`) ? 'Security Configuration' : cat?.displayName);
            // Get child category configuration
            this.getConfig(cat);
 });
```

In the above code we are filtering the security and advance categories and building group name based on the key



> What if the display name is missing?
key will always be there 
```
{"categories": [{"key": "omfAdvanced", "description": "", "displayName": "omfAdvanced"}]}

```